cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs

```diff
@@ -513,24 +515,24 @@ pKeyCmd =
             <> Opt.help "Use a Byron-era genesis delegate key, in legacy SL format."
             )
 
-    pByronKeyFile :: Parser SomeKeyFile
+    pByronKeyFile :: Parser (SomeKeyFile In)
     pByronKeyFile =
           (ASigningKeyFile      <$> pByronSigningKeyFile)
       <|> (AVerificationKeyFile <$> pByronVerificationKeyFile)
 
-    pByronSigningKeyFile :: Parser SigningKeyFile
+    pByronSigningKeyFile :: Parser (SigningKeyFile In)
```

Comment:
We can now be explicit in the types for parsers to say this is meant for input only.  If we use it for `Out` or for `InOut` or for the wrong content type, we get a type error.